### DISCUSSION TITLE
CPU profiling for AntMedia server on Ubuntu

### DISCUSSION THREAD

**Discussion by hd51515** [ant-media/Ant-Media-Server#5115] [General]
Are there any publicly available scripts/tools that can be used to profile on Ubuntu. 
It looks like chrome has some tools like pperf but I am not sure they work as is on the antmedia stack.
There is also perf tool usage documentation in chromium as shown below
perf record -g -a -e cycles -

**Comment by muratugureminoglu** [comment]
You can use `perf` tool on Ubuntu for performance analysis.

Install perf tool: 
```
apt-get install linux-tools-generic linux-tools-$(uname -r)
```
Here is an example:

1. to trace as follows
```
perf record -g /usr/local/antmedia/start.sh -e -a
```
2. send some loads 

Example with FFmpeg:
```
ffmpeg -re -i /your_video.mp4 -c copy -f flv rtmp://your_ams/WebRTCAppEE/stream1
```
3. report it as follows.

```
perf report
```

**Comment by hd51515** [comment]
Thanks
